### PR TITLE
Add: ability to delete vault

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Add endpoint, logic, component to be able to delete a vault.
Refactored to move logic from components to hooks.

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/data_source/DeleteStaticDataSourceDialog.tsx [hunk 1]
```diff
@@ -20,15 +25,15 @@ export function DeleteStaticDataSourceDialog({
 
   const message =
     dataSourceUsage === undefined
-      ? "Are you sure you want to permanently delete this data source?"
+      ? `Are you sure you want to permanently delete this ${getDataSourceName(dataSource)}?`
```
Comment (flvndvd) [inline front/components/data_source/DeleteStaticDataSourceDialog.tsx:28]: Shouldn't we use the folder's name, here?

Comment (flvndvd) [inline front/components/data_source/DeleteStaticDataSourceDialog.tsx:28]: `nit`
Extract `getDataSourceName(dataSource)` in a variable.

File: front/components/vaults/CreateOrEditVaultModal.tsx [hunk 2]
```diff
@@ -168,59 +171,45 @@ export function CreateOrEditVaultModal({
...
+  ]);
 
-  const rows = useMemo(() => getTableRows(members), [members]);
+  const onDelete = useCallback(async () => {
+    if (!vault) {
+      return;
+    }
 
-  const columns = useMemo(() => getTableColumns(), [getTableColumns]);
+    const res = await doDelete(vault);
```
Comment (flvndvd) [inline front/components/vaults/CreateOrEditVaultModal.tsx:207]: I'm wondering if this `doDelete` shouldn't be in the modal itself, WDYT?

File: front/components/vaults/CreateOrEditVaultModal.tsx [hunk 3]
```diff
@@ -312,6 +269,25 @@ export function CreateOrEditVaultModal({
...
+            <>
+              <Page.Separator />
+              <ConfirmDeleteVaultDialog
+                vault={vault}
+                handleDelete={onDelete}
+                isOpen={showDeleteConfirmDialog}
+                onClose={() => setShowDeleteConfirmDialog(false)}
+              />
+              <div className="flex w-full justify-end">
+                <Button
```
Comment (flvndvd) [inline front/components/vaults/CreateOrEditVaultModal.tsx:282]: We've discussed this broadly last week but we agreed to add the button in the modal, and the modal takes care of handling it's own state.

File: front/components/vaults/CreateOrEditVaultModal.tsx [hunk 4]
```diff
@@ -312,6 +269,25 @@ export function CreateOrEditVaultModal({
               </div>
             )}
           </div>
+          {vault && (
```
Comment (flvndvd) [inline front/components/vaults/CreateOrEditVaultModal.tsx:272]: Not all vaults can be deleted. Can we only display if regular vault?

File: front/lib/api/vaults.ts [hunk 5]
```diff
@@ -1,18 +1,43 @@
...
-
-  if (isWebsite(dataSourceResource)) {
-    return "website";
+  if (!vault.isRegular()) {
+    throw new Error("Cannot delete non regular vaults.");
   }
 
-  return "managed";
+  const dataSourceViews = await DataSourceViewResource.listByVault(auth, vault);
+  // Check if any data source view is in use in a agent configuration
```
Comment (flvndvd) [inline front/lib/api/vaults.ts:18]: Is this a TODO? If so, let's use `// TODO(VAULTS_INFRA xxx)`

File: front/lib/resources/data_source_view_resource.ts [hunk 6]
```diff
@@ -32,6 +33,20 @@ import {
 import type { ResourceFindOptions } from "@app/lib/resources/types";
 import type { VaultResource } from "@app/lib/resources/vault_resource";
 
+export const getDataSourceCategory = (
```
Comment (flvndvd) [inline front/lib/resources/data_source_view_resource.ts:36]: This code already exist here: https://github.com/dust-tt/dust/blob/f66d3788c68cca3d30d67d005f1e3d5ece9bd844/front/lib/api/vaults.ts#L6-L18

File: front/lib/resources/group_resource.ts [hunk 7]
```diff
@@ -629,9 +627,9 @@ export class GroupResource extends BaseResource<GroupModel> {
         transaction,
       });
 
-      await this.model.destroy({
+      await GroupMembershipModel.destroy({
```
Comment (flvndvd) [inline front/lib/resources/group_resource.ts:630]: So we don't delete `GroupModel` anymore 🤔

File: front/lib/vaults.ts [hunk 8]
```diff
@@ -17,3 +17,22 @@ export function getVaultIcon(
 export const getVaultName = (vault: VaultType) => {
   return vault.kind === "global" ? "Company Data" : vault.name;
 };
+
+import type { DataSourceViewCategory } from "@dust-tt/types";
+
+import { isFolder, isWebsite } from "@app/lib/data_sources";
+import type { DataSourceResource } from "@app/lib/resources/data_source_resource";
+
+export const getDataSourceCategory = (
```
Comment (flvndvd) [inline front/lib/vaults.ts:26]: Impressive you managed to write twice the same function in the same PR for a function that already existed 😅 🤯 !

File: front/components/vaults/CreateOrEditVaultModal.tsx (hunk 2 above)
Comment (Fraggle) [reply front/components/vaults/CreateOrEditVaultModal.tsx:207]: let's revisit in another PR if you don't mind.

File: front/components/vaults/CreateOrEditVaultModal.tsx (hunk 3 above)
Comment (Fraggle) [reply front/components/vaults/CreateOrEditVaultModal.tsx:282]: let's revisit in another PR if you don't mind.

File: front/lib/resources/data_source_view_resource.ts [hunk 9]
```diff
@@ -32,6 +33,20 @@ import {
 import type { ResourceFindOptions } from "@app/lib/resources/types";
 import type { VaultResource } from "@app/lib/resources/vault_resource";
 
+const getDataSourceCategory = (
```
Comment (flvndvd) [inline front/lib/resources/data_source_view_resource.ts:36]: Why not using the one from here?

https://github.com/dust-tt/dust/blob/main/front/lib/api/vaults.ts

**Review by flvndvd — APPROVED** [review]

File: front/lib/resources/data_source_view_resource.ts (hunk 9 above)
Comment (Fraggle) [reply front/lib/resources/data_source_view_resource.ts:36]: cycle dependency.
If you look closely, you'll notice I removed it.

**Pull request merged**